### PR TITLE
6000 remove zoe floating promises

### LINKED ISSUE #6000: resolve all floating promise warnings
## What is the Problem Being Solved?

Since https://github.com/Agoric/agoric-sdk/pull/5495 we have tooling that will detect and warn on floating promises.

https://github.com/Agoric/agoric-sdk/commit/4a84690f226c2e20bce25629757de4113f79d931 disabled them in CI as too slow. We've elected to postpone [re-enabling them in CI](https://github.com/Agoric/agoric-sdk/issues/5788) until after launch. This is fine because we don't necessarily need the feedback on every run.

The problem at hand is that we have unhandled (floating) promises in master that may contain bugs. We've seen them _hide_ bugs by creating bugs in tests.

## Description of the Design

Before launch, handle all `no-floating-promises` warnings.

Tackle on a per-package basis by running,
```
NODE_OPTIONS='--max-old-space-size=8192' AGORIC_ESLINT_TYPES=true yarn lint:eslint
```

https://github.com/Agoric/agoric-sdk/issues/5535 is an example. Though note it used `scripts/lint-with-types.sh` which since 4a84690f226c2e20bce25629757de4113f79d931 is a misnomer; it's more `ci-lint.sh` because it handles splitting the lint job and configuration of whether type linting is enabled in CI.

## Security Considerations
--
## Test Plan

Running `AGORIC_ESLINT_TYPES=true yarn lint | grep @typescript-eslint/no-floating-promises` has no output.

### PR DESCRIPTION
**Pull request by Chris-Hibbert**
closes: #6000

## Description

Resolve remaining floating promises in Zoe

### Security Considerations

Zoe's correct functioning is crucial to the chain. I determined that it was correct to not further handle any of these promises.

### Scaling Considerations

N/A

### Documentation Considerations

N/A

### Testing Considerations

N/A

### Upgrade Considerations

No impact, because no behavioral changes.

### CODE REVIEW

**Comment by Chris-Hibbert** [comment]
@turadg: please check out my revisions to the tests, particularly `priceAggregator test`. The commits segregate the changes cleanly.

File: packages/zoe/test/swingsetTests/brokenContracts/vat-alice.js [hunk 1]
```diff
@@ -42,7 +41,7 @@ const build = async (log, zoe, issuers, payments, installations) => {
     });
     const alicePayments = { Asset: moolaPayment };
 
-    logCounter(log, publicFacet);
+    await logCounter(log, publicFacet);
```
Comment (turadg) [inline packages/zoe/test/swingsetTests/brokenContracts/vat-alice.js:44]: probably ok if these stayed `void` but good to have the confident now of what the order is

File: packages/zoe/test/unitTests/contracts/priceAggregator.test.js [hunk 2]
```diff
@@ -732,15 +737,20 @@ test('quoteWhen', async t => {
 
   /** @type {PriceQuote | undefined} */
   let abovePriceQuote;
-  t.notThrowsAsync(quoteWhenGTE.then(result => (abovePriceQuote = result)));
+  const results = [];
```
Comment (turadg) [inline packages/zoe/test/unitTests/contracts/priceAggregator.test.js:745]: consider commenting here why this array is necessary

**Review by turadg — APPROVED** [review]
It's a relief to see that the only `awaits` necessary were in test code. =)

**Pull request merged**